synapse/handlers/sso.py

```diff
@@ -710,7 +712,7 @@ async def _register_mapped_user(
         # Since the localpart is provided via a potentially untrusted module,
         # ensure the MXID is valid before registering.
         if not attributes.localpart or contains_invalid_mxid_characters(
-            attributes.localpart
+            attributes.localpart, self._e164_mxids
         ):
```

Comment:
The SSO mapping code (done via the `map_username_to_mxid_localpart`) doesn't really support this though, so maybe we should hard-code the calls in this file to `False`?

Reply:
Note that we don't need to worry about changing a currently registered user's mapping since we find those users via their remote ID.

Changing the mapping could, however, cause conflicts where there didn't used to be one.